ParameterDataApi.yaml

```diff
@@ -88,122 +144,79 @@ components:
+          type: object
+          additionalProperties: true    
+        
       required:
         - timestamp
-        - fields
+        - data    
 
-    DataType:
-      type: string
-      enum:
-        - NUMBER
-        - INTEGER
-        - STRING
-        
     HeaderDefinition:
       type: object
       properties: 
-        startTime:
-          description: The start time for the time series
+      
+        componentUuid:
           type: string
-          format: date-time
-        
-        endTime: 
-          description: The end time for the time series
+          format: uuid
+          description: The unique identifier of the component within FRAIHMWORK that the parameter data associates with
+      
+        messageType:
+          type: string          
+          description: The type of the message, dictated by the client for delineating individual message formats
+
+        messageVersion:
           type: string
-          format: date-time
+          description: The version string for the parameter data messageType. If the client changes the data fields within a message stream, they should increment the version accordingly to distinguish message format along with the messageType
           
-        columns:
-          $ref: '#/components/schemas/ColumnDefinition'
-        
       required:
-        - startTime
-        - endTime
-        - columns
+        - componentUuid
+        - messageType
+        - version
 
-    TimeSeriesDataDefinition:
+    SingleDataDefinition:
       type: object
-      description: A definition of time series data. This is loosely based upon Json Time Series, Version 1.0. (https://docs.eagle.io/en/latest/reference/index.html)
+      description: A definition of a set of parameter data having a single discrete timestamp. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys
       properties:
-        docType:
-          type: string          
-          description: The type of the document
-
-        version:
-          type: string
-          description: The version string for the time series data definition
 
         header:
           $ref: '#/components/schemas/HeaderDefinition'
 
-        data:
-          $ref: '#/components/schemas/DataDefinition'
-
+        single:
+          $ref: '#/components/schemas/DataSample'
+         
       required: 
-        - docType
         - header
-        - data
+        - single
 
-    ValueSpec:
+    TimeSeriesDataDefinition:
       type: object
+      description: A time series ordered collection for a set of parameter data. For multiple parameter observations for the given timestamp, use multiple key/value pairs with unique keys. It is encouraged that each element in the array has the same keys in the same order
```

Comment:
This description implies that there is a requirement that the data be ordered by time in the array. If it isn't, is that grounds for a 400 return on the API call? Or do we accept and sort for them?

Is also worth mentioning that it doesn't mention which direction the ordering should be. A case could be made for ascending or descending timestamp ordering (I would interpret the lack of direction as ascending, but people be cray-cray)

Reply:
Good comment. The part of it that applies to the API spec is the stating the order. Sorting for them would be a good thing to do. Time series data is supposed to be in order of occurrence.